packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -232,15 +241,8 @@ async function getIslandContext (event: H3Event): Promise<NuxtIslandContext> {
   return ctx
 }
 
-const HAS_APP_TELEPORTS = !!(appTeleportTag && appTeleportId)
-const APP_TELEPORT_OPEN_TAG = HAS_APP_TELEPORTS ? `<${appTeleportTag} id="${appTeleportId}">` : ''
-const APP_TELEPORT_CLOSE_TAG = HAS_APP_TELEPORTS ? `</${appTeleportTag}>` : ''
-
-const APP_ROOT_OPEN_TAG = `<${appRootTag}${appRootId ? ` id="${appRootId}"` : ''}>`
-const APP_ROOT_CLOSE_TAG = `</${appRootTag}>`
-
 const PAYLOAD_URL_RE = process.env.NUXT_JSON_PAYLOADS ? /\/_payload.json(\?.*)?$/ : /\/_payload.js(\?.*)?$/
-const ROOT_NODE_REGEX = new RegExp(`^${APP_ROOT_OPEN_TAG}([\\s\\S]*)${APP_ROOT_CLOSE_TAG}$`)
+const ROOT_NODE_REGEX = new RegExp(`^<${appRootTag}[^>]*id\\s*=\\s*["']${appRootId}["'][^>]*>([\\s\\S]*)<\\/${appRootTag}>$`)
```

Comment:
This regex needs a look, it requires an id to be set for the root tag.

Reply:
I think we don't really need the ID - we're just stripping the outer tag from the HTML within this line:

https://github.com/nuxt/nuxt/blob/e9181116a3e25f5507ba1df11ef5515ed05cf297/packages/nuxt/src/core/runtime/nitro/renderer.ts#L637-L643

Which means, I think, that we can be pretty imprecise on the RE.

Reply:
There can be multiple multiple root (body > div) tags both prepended and appended from the "root" node? This logic wants only the "root", not the teleports right or the other appended / prepended Unhead templates :thinking: 

https://github.com/nuxt/nuxt/blob/e9181116a3e25f5507ba1df11ef5515ed05cf297/packages/nuxt/src/core/runtime/nitro/renderer.ts#L561

https://github.com/nuxt/nuxt/blob/e9181116a3e25f5507ba1df11ef5515ed05cf297/packages/nuxt/src/core/runtime/nitro/renderer.ts#L471-L473

Reply:
The way the regexp is written, it only looks at the first item in `html.body` (not html.bodyPrepend, or the other bits), and it intends to match the entire string (hence the `^` and `$`).

I don't think it would work (as written currently) if the user prepended an item in `body` within the `render:html` hook. (But if the user doesn't do that, we know for sure it will be just the result from `renderToString().html`.)